app/src/org/commcare/utils/FileUtil.java

```diff
@@ -724,13 +724,12 @@ public static void copyFile(InputStream inputStream, File dstFile) throws IOExce
         outputStream.close();
     }
 
-    public static boolean isSupportedMultiMediaFile(Uri media) {
+    public static boolean isSupportedMultiMediaFile(Context context, Uri media) {
         try {
-            String binaryPath = UriToFilePath.getPathFromUri(CommCareApplication.instance(), media);
-            return FormUploadUtil.isSupportedMultimediaFile(binaryPath);
-        } catch (UriToFilePath.NoDataColumnForUriException e) {
-            // No file path available, work with the media uri instead
-            return FormUploadUtil.isSupportedMultimediaFile(media.getPath());
+            String fileName = getFileName(context, media);

```

Comment:
It feels like `UriToFilePath.getPathFromUri` handles a lot of edge cases around media URIs that we have encountered over the years. Is there a reason we think `getFileName` is better here ? If not, Is it possible to use `getFileName` as an alternative in cases we don't get an extension using `UriToFilePath.getPathFromUri`

Reply:
`getFileName` also uses `UriToFilePath` [here](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/utils/FileUtil.java#L537-L541), but it only executes when the contentResolver couldn't find the display name. 

I'm not sure if `UriToFilePath` gives a better result than contentResolver, but we can improve that and add a case where if the fileName returned by contentResolver doesn't have an extension, we should grab fileName from `UriToFilePath`. Does that sound good?

Reply:
I see, agree with your approach here.